django/db/models/sql/compiler.py

```diff
@@ -1836,7 +1836,23 @@ def pre_sql_setup(self):
         query.clear_ordering(force=True)
         query.extra = {}
         query.select = []
-        query.add_fields([query.get_meta().pk.name])
+        meta = query.get_meta()
+        fields = [meta.pk.name]
+        related_ids_index = []
+        for related in self.query.related_updates:
+            if all(
+                path.join_field.primary_key for path in meta.get_path_to_parent(related)
+            ):
+                # If a primary key chain exists to the targeted related
+                # update the `meta.pk` value can be used for it.
+                related_ids_index.append((related, 0))
```

Comment:
This ensures the code stays unchanged in cases model inheritance is used but not MTI.